airbyte-lib/airbyte_lib/results.py

```diff
@@ -15,28 +15,39 @@
 
 
 class ReadResult:
-    def __init__(self, processed_records: int, cache: SQLCacheBase) -> None:
+    def __init__(
+        self, processed_records: int, cache: SQLCacheBase, processed_streams: list[str]
+    ) -> None:
         self.processed_records = processed_records
         self._cache = cache
+        self._processed_streams = processed_streams
 
     def __getitem__(self, stream: str) -> CachedDataset:
-        if stream not in self._cache:
+        if stream not in self._processed_streams or stream not in self._cache:
             raise KeyError(stream)
 
         return CachedDataset(self._cache, stream)
 
     def __contains__(self, stream: str) -> bool:
-        return stream in self._cache
+        return stream in self._cache and stream in self._processed_streams
```

Comment:
Should we just check if it is in self._processed_streams? It seems that anything that is in processed streams list should also be in the cache (in theory at least).